assets/scripts/check-deeplink.js

```diff
@@ -0,0 +1,133 @@
+function _createHiddenIframe(target, uri) {
+  const iframe = document.createElement("iframe");
+  iframe.src = uri;
+  iframe.id = "hiddenIframe";
+  iframe.style.display = "none";
+  target.appendChild(iframe);
+  return iframe;
+}
+
+function openUriWithHiddenFrame(uri, failCb) {
+  const timeout = setTimeout(function() {
+    failCb();
+    handler.remove();
+  }, 500);
+
+  let iframe = document.querySelector("hiddenIframe");
+  if (!iframe) {
+    iframe = _createHiddenIframe(document.body, "about:blank");
+  }
+
+  const handler = window.addEventListener("blur", onBlur);
+  function onBlur() {
+    clearTimeout(timeout);
+    handler.remove();
+  }
+
+  iframe.contentWindow.location.href = uri;
+}
+
+function openUriWithTimeoutHack(uri, failCb) {
+  const timeout = setTimeout(function() {
+    failCb();
+    handler.remove();
+  }, 500);
+
+  //handle page running in an iframe (blur must be registered with top level window)
+  let target = window;
+  while (target != target.parent) {
+    target = target.parent;
+  }
+
+  const handler = target.addEventListener("blur", onBlur);
+  function onBlur() {
+    if (handler) {
+      clearTimeout(timeout);
+      handler.remove();
+    }
+  }
+
+  window.location = uri;
+}
+
+function openUriWithMsLaunchUri(uri, failCb) {
+  navigator.msLaunchUri(uri, undefined, failCb);
+}
+
+function checkBrowser() {
+  const isOpera = !!window.opera || navigator.userAgent.indexOf(" OPR/") >= 0;
+  const ua = navigator.userAgent.toLowerCase();
+  const isSafari =
+    (~ua.indexOf("safari") && !~ua.indexOf("chrome")) ||
+    Object.prototype.toString.call(window.HTMLElement).indexOf("Constructor") >
+      0;
+  return {
+    isOpera,
+    isFirefox: typeof InstallTrigger !== "undefined",
+    isSafari,
+    isIOS: /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream,
+    isChrome: !!window.chrome && !isOpera
+  };
+}
+
+function check(uri, failCb) {
+  function failCallback() {
+    failCb && failCb();
+  }
```

Comment:
What is the purpose of `failCallback`, as `check` is always called with a `failCb` function?